### ISSUE TITLE
Plugin published on Grafana.com - feedback

### ISSUE THREAD

**Issue by daniellee** [doitintl/bigquery-grafana#65]
I have just published the plugin on Grafana.com: https://grafana.com/plugins/doitintl-bigquery-datasource

Mostly it looks great, you took some of the best bits of the Stackdriver and MySQL/Postgres datasources. Great work!

I found some smaller items that could be improved:

- I tested with a service account that did not have the correct roles but the Save and Test function still said it was ok.
- Error handling. The Stackdriver and Sql datasource show errors in the query editor. At the moment, I would have to go to the Chrome console to see the error.
    
    ![image](https://user-images.githubusercontent.com/434655/58245641-8ba43f00-7d55-11e9-9f26-b032939d5f7e.png)
- Ran into problems with the macros when writing sql. The $__time macro and the $__timeGroup macros did not get interpolated sometimes (the timeFilter macro always gets interpolated):
    ```sql
    SELECT
      $__timeGroup(timestamp, '1m'),
      metric AS metric,
      value AS value
    FROM
      `my-production.GrafanaTest.GrafanaTestTable`
    WHERE
      $__timeFilter(timestamp)
    GROUP BY timestamp
    ORDER BY timestamp
    ```
    This is with a simple table with following schema:
    ![image](https://user-images.githubusercontent.com/434655/58246269-e5f1cf80-7d56-11e9-8d28-cf80cd665646.png)

If you want to update the version on Grafana.com then please send a PR to the Grafana plugin repo and add a new version to the version array here:

https://github.com/grafana/grafana-plugin-repository/blob/master/repo.json#L3047-L3058

Like this:

```json
{
      "id": "doitintl-bigquery-datasource",
      "type": "datasource",
      "url": "https://github.com/doitintl/bigquery-grafana",
      "versions": [
       {
          "version": "<new version x.x.x>",
          "commit": "<new commit sha>",
          "url": "https://github.com/doitintl/bigquery-grafana "
        },
        {
          "version": "0.5.1",
          "commit": "8a74d87e36ef6dfec58a813850419a368a9361d3",
          "url": "https://github.com/doitintl/bigquery-grafana "
        }
      ]
    }
```

In conclusion, thanks so much for building this! The Grafana community has been asking for this for ages: https://github.com/grafana/grafana/issues/7872 

Looking forward to seeing how this develops going forward.

**Comment by avivl** [comment]
@daniellee thank you for your feedback!

1. I tested with a service account that did not have the correct roles but the Save and Test function still said it was ok. - **_I'll fix this_** 

2. Errors should be displayed on the left upper corner in red. If this is not the case, please LMK

3. I'm not sure I understand the issue with the macros. Could you also please the resulting SQL (it's in the console)

4. Thank you for putting it on Grafana.com. I'll make sure to create a PR with every new version

Thanks again
Aviv

**Comment by daniellee** [comment]
When it doesn't work the resulting sql looks like this (you see that the first macro did not get replaced):

```sql
SELECT
  $__timeGroup(timestamp, '1m'),
  metric AS metric,
  value AS value
FROM
  `my-production.GrafanaTest.GrafanaTestTable`
WHERE
  timestamp BETWEEN '2019-05-23 04:51:29' AND '2019-05-23 16:51:29'
GROUP BY timestamp
ORDER BY timestamp LIMIT 413
```
This could absolutely be me doing something stupid.

**Comment by avivl** [comment]
@daniellee  the correct syntax is $__timeGroup(timestamp,1m) - No space and no qoutation

**Comment by daniellee** [comment]
@avivl Thanks - can confirm that it works when removing the quotes and the space. But this is definitely a bug as `$__timeGroup(timestamp, '1m')` is what the query editor generates when switching from the visual mode to the raw mode. Or if I create a new query and don't change anything then it generates a query with `$__time(time_column)` that does not get interpolated. Maybe this [default query](https://github.com/doitintl/bigquery-grafana/blob/master/src/query_ctrl.ts#L13) in the queryCtrl is incorrect?

I would also say that the datasource should be able to handle spaces between parameters in macros.

Do you want me to open a new issue(s) for these?

**Comment by avivl** [comment]
@daniellee Yes thank you!

**Comment by avivl** [comment]
#77

**Comment by avivl** [comment]
@daniellee This query (Maybe this default query in the queryCtrl is incorrect?) is a place holder. When you create an empty query we do not know yet anything (project, dataset table etc)

**Comment by chompysticks** [comment]
Query LIMIT doesn't change when changing time ranges from Quick ranges dashboard dropdown. 

I found when setting to last 1 hour the LIMIT query option is added (looking at Query Inspector output), and in my case it's set to LIMIT 821 but when I change the time range to for example last 24 not all data is returned from the query because the LIMIT is still set to 821.

I'm on Grafana cloud and the plugin version is 5.1.

Also would like to point out that the cloud plugin version 5.1 has the Table bug "undefined is not an object (evaluating 'm.target.search')". Is it possible to update the cloud version to 0.6.0 please that resolves this bug too ?

**Comment by avivl** [comment]
@chompysticks The LIMIT has nothing to do with the time range.
It is determinate by the Max data points parameters which the browser set. (You can override it)
I submitted a pull request for GRfana to update the plugin for it's the latest version  a more than a week ago

**Comment by chompysticks** [comment]
@avivl cheers. I hope they update it soon :(.
